### DISCUSSION TITLE
Is it possible to specify generator?

### DISCUSSION THREAD

**Discussion by fralik** [scikit-build/scikit-build-core#1465] [Q&A] [answered]
I wonder if it is possible to specify generator via `pyproject.toml` file? Or shall we rely on auto-detection of the most suitable generator? I am on Windows, would like to use Ninja, but VS Studio is keep being selected.

**Comment by henryiii** [comment, marked as answer]
You can set `cmake.args = ["-GNinja"]`. If you want to customize this per-platform, you can use `[[tool.scikit-build.overrides]]`. Now you can even use `inherit` to build them up. However, ninja's not as well supported quite yet on Windows; we won't detect cmake.args's -G when computing get_requries_* (MSVC ships with ninja, so that's pretty safe), and we won't look in the registry and activate MSVC for you, so it needs to be done in a dev console or after activating vcvars.bat.

**Reply by fralik** [reply]
Thanks!
